tests/h/search/conftest.py

```diff
@@ -56,7 +56,9 @@ def _index(*annotations):
 
 @pytest.fixture
 def search_index(es_client, pyramid_request, moderation_service):
-    return SearchIndexService(es_client, pyramid_request)
+    return SearchIndexService(
+        pyramid_request, es_client, session=pyramid_request.db, settings={}
```

Comment:
A very handy (but unintentional?) property of the settings service is it exports `get()` as basically it's only method. That means a `dict` will duck type as the settings service.

Reply:
Although using an autospecced mock for the settings service as means that it'll update itself if the settings service's interface changes

Reply:
You can't use an autospecced one. We really need this to work rather than be mocked out. This isn't to test the `SearchIndexService` but to use it to setup fixtures for the other objects. We actually use it to index. 

The settings here are only checked to see if we are in re-index mode, which we never will be here. This is the equivalent of the previous `index()` calls.